MoneyTransferWorker/Workflow.cs

```diff
@@ -0,0 +1,58 @@
+namespace MoneyTransferProject;
+using Temporalio.Workflows;
+using Temporalio.Common;
+
+[Workflow]
+public class MoneyTransferWorkflow 
+{
+    [WorkflowRun]
+    public async Task<string> RunAsync(PaymentDetails details)
+    {
+        // Retry policy 
+        var retryPolicy = new RetryPolicy
+        {
+            InitialInterval = TimeSpan.FromSeconds(1),
+            MaximumInterval = TimeSpan.FromSeconds(100),
+            BackoffCoefficient = 2,
+            MaximumAttempts = 5,
+            NonRetryableErrorTypes = new[] { "InvalidAccountError", "InsufficientFundsError" }
+        };
+
+        string withdrawResult;
+        try
+        {
+            withdrawResult = await Workflow.ExecuteActivityAsync<string>(
+                "Withdraw", // registered activity name
+                new object[] { details },
+                new ActivityOptions { ScheduleToCloseTimeout = TimeSpan.FromMinutes(5), RetryPolicy = retryPolicy }
+            );
+        }
+        catch (ApplicationException ex) when (ex.InnerException is InsufficientFundsError)
+        {
+            throw new ApplicationException("Withdrawal failed due to insufficient funds.", ex);
+        }
```

Comment:
Can probably just let this bubble out instead of wrapping since you're not giving an error type or other discriminator for the caller to use anyways, but your choice